src/main/java/frc/robot/subsystems/Shooter/commands/PIDshoot.java

```diff
@@ -0,0 +1,20 @@
+package frc.robot.subsystems.Shooter.commands;
+
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.Constants;
+import frc.robot.subsystems.Shooter.Shooter;
+
+/**
+ * this command activates the shooter subsystem based on distance
+ */
+public class PIDshoot extends CommandBase {
+    private Shooter shooter;
+    private double distance;
+    private double power;
+
+    public PIDshoot(Shooter shooter, double distance, double power) {

```

Comment:
Why do you take power as an arguement? You should use function of velocity base on distance, and then calulate desired velocity and use `ControlMode.Velocity`